src/Microsoft.Health.Fhir.CosmosDb/Features/Operations/Reindex/CosmosBackgroundJobThrottleController.cs

```diff
@@ -30,40 +34,52 @@ public class ReindexJobCosmosThrottleController : IReindexJobThrottleController
         private uint _jobConfiguredBatchSize = 100;
         private bool _initialized = false;
         private readonly RequestContextAccessor<IFhirRequestContext> _fhirRequestContextAccessor;
-        private readonly ILogger<ReindexJobCosmosThrottleController> _logger;
+        private readonly Func<IScoped<IFhirDataStore>> _fhirDataStoreFactory;
+        private readonly ILogger<CosmosBackgroundJobThrottleController> _logger;
 
-        public ReindexJobCosmosThrottleController(
+        public CosmosBackgroundJobThrottleController(
             RequestContextAccessor<IFhirRequestContext> fhirRequestContextAccessor,
-            ILogger<ReindexJobCosmosThrottleController> logger)
+            Func<IScoped<IFhirDataStore>> fhirDataStoreFactory,
+            ILogger<CosmosBackgroundJobThrottleController> logger)
         {
-            EnsureArg.IsNotNull(fhirRequestContextAccessor, nameof(fhirRequestContextAccessor));
-            EnsureArg.IsNotNull(logger, nameof(logger));
-
-            _fhirRequestContextAccessor = fhirRequestContextAccessor;
-            _logger = logger;
+            _fhirRequestContextAccessor = EnsureArg.IsNotNull(fhirRequestContextAccessor, nameof(fhirRequestContextAccessor));
+            _fhirDataStoreFactory = EnsureArg.IsNotNull(fhirDataStoreFactory, nameof(fhirDataStoreFactory));
+            _logger = EnsureArg.IsNotNull(logger, nameof(logger));
         }
 
-        public ReindexJobRecord ReindexJobRecord { get; set; } = null;
+        public IThrottleableJobRecord JobRecord { get; set; } = null;
 
-        public void Initialize(ReindexJobRecord reindexJobRecord, int? provisionedDatastoreCapacity)
+        public async Task Initialize(IThrottleableJobRecord jobRecord, CancellationToken cancellationToken)
         {
-            EnsureArg.IsNotNull(reindexJobRecord, nameof(reindexJobRecord));
+            EnsureArg.IsNotNull(jobRecord, nameof(jobRecord));
+
+            if (jobRecord.TargetDataStoreUsagePercentage != null &&
+                    jobRecord.TargetDataStoreUsagePercentage > 0)
+            {
+                using (IScoped<IFhirDataStore> store = _fhirDataStoreFactory.Invoke())
+                {
+                    _provisionedRUThroughput = await store.Value.GetProvisionedDataStoreCapacityAsync(cancellationToken);
```

Comment:
@feordin is this the call that doesn't work in paas? I also don't think we can read these values under Cosmos/MI

Reply:
@brendankowitz I think you are correct that with MI we would need a control pane role to access this information, at least a read-only role.  I cannot remember the details with this method in PaaS, if our Comsos DB access key allowed permission to read  this information.

Reply:
Is this something that isn't working right now? This code is what was already there, I just moved it around.

Reply:
I've reverted all the throttling changes. The deletion service already has a retry on the delete, but it doesn't look like there is any retry on the search at any level. Due to the high usage I've seen for this service I'm going to increase the backoff for the retry in the deletion service.